bids/layout/bids_validator.py

```diff
@@ -252,3 +256,142 @@ def conditional_match(self, expression, path):
+    --------
+    >>> layout = bids.grabbids.BIDSLayout('/path_to/sample_project_root')
+    >>> duplicate_file_df = check_duplicate_files(layout)
+    >>> duplicate_file_df
+    # Put example output here
+
+
+    Notes
+    ------
+    Returns a data frame in which the first column is the file
+    identifier and the second column is the path to the file.
+    Files with matching identifiers have the same content.
+    """
+    
+    def md5(fname):
+        hash_md5 = hashlib.md5()
+        with open(fname, "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                hash_md5.update(chunk)
+        return hash_md5.hexdigest()
+    hash_map = {}
+    all_niftis = layout.get(return_type="file", extensions='.nii.gz')
+    for nifti_file in all_niftis:
+        md5sum = md5(nifti_file)
+        if md5sum in hash_map:
+            hash_map[md5sum].append(nifti_file)
+        else:
+            hash_map[md5sum] = [nifti_file]
+    df = pd.DataFrame.from_dict(hash_map, orient='index') 
+    pruned_df = df.stack().reset_index().drop(columns='level_1')
+    out_df = pruned_df.rename(columns={'level_0': 'hash', 0: 'filename'})
+    return out_df
+    
+    
+def check_expected_files(layout, config):
+    """Checks files in BIDS project match user defined expectations.
+
+    This method checks the number of data set files against a user customized 
+    configuration file. Returns two data frames: summary_df, problem_df.
+
+
+    Parameters
+    ----------
+        layout: BIDSLayout class
+            A BIDSLayout path of a data set.
+
+        config: string
+            Path to customized configuration file.
+
+
+    Examples
+    --------
+    >>> layout = bids.grabbids.BIDSLayout('/path_to/sample_project_root')
+    >>> summary_df, problem_df = check_expected_files(layout, 'pybids/bids/layout/tests/data/sample_validation_config.json')
+    >>> summary_df
+    # Put example output here
+    >>> problem_df
+    # Put example output here
+
+
+    Notes
+    --------
+    
+    The configuration file can take any keys that are valid arguments for
+    pybids `layout.get()` Values shoud match those in the BIDS file names. 
+    See the sample config for an example (bids/layout/tests/data/sample_validation_config.json).
+    The more specific keys are provided, the more informative the output will be.
+
+    """
+
+    dictlist = []
+    with open(config) as f:
+        json_data = json.load(f)
+        subjects = layout.get_subjects()
+    for sub in subjects: 
+        for scan_params_d in json_data['sequences']:
+            scan_params = scan_params_d.copy()
+            seq_params = {i: scan_params[i] for i in scan_params if i != 'runs'}
+            actual_runs = layout.get(return_type='obj', subject=sub, extensions='.nii.gz', **seq_params)
+            scan_params['subject'] = sub
+            scan_params['runs_found'] = len(actual_runs)
+            scan_params['problem'] = len(actual_runs) != scan_params['runs']
```

Comment:
If `'runs'` isn't defined in the config file, this will fail... either add a conditional to check `if 'runs' in scan_params` before running this line, or state that `'runs'` is mandatory in the docstring description of the config.